src/anomalib/trainer/utils/thresholder.py

```diff
@@ -86,17 +73,45 @@ def update(self, outputs: STEP_OUTPUT) -> None:
         Args:
             outputs (STEP_OUTPUT): Step outputs.
         """
-        if self.threshold_method == ThresholdMethod.ADAPTIVE:
-            self._update_thresholds(self.trainer.image_threshold, self.trainer.pixel_threshold, outputs)
-
-    @staticmethod
-    def _update_thresholds(
-        image_metric: AnomalyScoreThreshold,
-        pixel_metric: AnomalyScoreThreshold,
-        outputs: STEP_OUTPUT,
-    ) -> None:
+        image_metric = self.trainer.image_threshold
+        pixel_metric = self.trainer.pixel_threshold
         image_metric.cpu()
         image_metric.update(outputs["pred_scores"], outputs["label"].int())
-        if "mask" in outputs.keys() and "anomaly_maps" in outputs.keys():
+        if (
+            self.trainer.task_type != TaskType.CLASSIFICATION
+            and "anomaly_maps" in outputs.keys()
+            and "mask" in outputs.keys()
+        ):
             pixel_metric.cpu()
+            # TODO this should use bounding boxes for detection task type
             pixel_metric.update(outputs["anomaly_maps"], outputs["mask"].int())
+
+    def _get_threshold_method(self, threshold_method: dict | None) -> BaseAnomalyScoreThreshold:
+        """Get threshold method.
+
+        Args:
+            threshold_method (dict | None): Threshold method. Defaults to AdaptiveScoreThreshold.
+
+        Returns:
+            Instantiated threshold method.
+        """
+        thresholder: BaseAnomalyScoreThreshold
```

Comment:
Following the naming discussion above, this variable could be called `threshold_metric`